batavia/core/builtins.js

```diff
@@ -732,8 +732,9 @@ batavia.builtins.help.__doc__ = 'In Python, this is a wrapper around pydoc.help.
 batavia.builtins.hex = function(args) {
     if (args.length !== 1) {
         throw new batavia.builtins.TypeError("hex() takes exactly one argument (" + args.length + " given)");
-    }
-    return "0x" + args[0].toString(16);
+    };
+    var int = args[0].val.c[0]
```

Comment:
Patch incoming; no, it wasn't tested. It is now, with 0xf, 0xdeadbeef, and a really big int.